algoliasearch/index.go

```diff
@@ -770,3 +759,87 @@ func (i *index) SearchRulesWithRequestOptions(params Map, opts *RequestOptions)
 	err = i.client.request(&res, "POST", path, params, read, opts)
 	return
 }
+
+// ReplaceAllSynonyms replace all the synonyms of the current index with the given ones.
+func (i *index) ReplaceAllSynonyms(synonyms []Synonym) (res UpdateTaskRes, err error) {
+	return i.ReplaceAllSynonymsWithRequestOptions(synonyms, nil)
+}
+
+// ReplaceAllSynonymsWithRequestOptions is the same as ReplaceAllSynonyms but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllSynonymsWithRequestOptions(synonyms []Synonym, opts *RequestOptions) (res UpdateTaskRes, err error) {
+	return i.BatchSynonymsWithRequestOptions(synonyms, true, true, opts)
+}
+
+// ReplaceAllRules replace all the rules of the current index with the given ones.
+func (i *index) ReplaceAllRules(rules []Rule) (res BatchRulesRes, err error) {
+	return i.ReplaceAllRulesWithRequestOptions(rules, nil)
+}
+
+// ReplaceAllRulesWithRequestOptions is the same as ReplaceAllRules but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllRulesWithRequestOptions(rules []Rule, opts *RequestOptions) (res BatchRulesRes, err error) {
+	return i.BatchRulesWithRequestOptions(rules, true, true, opts)
+}
+
+// ReplaceAllObjects replace all the objects of the current index with the given ones.
+func (i *index) ReplaceAllObjects(objects []Object) error {
+	return i.ReplaceAllObjectsWithRequestOptions(objects, nil)
+}
+
+// ReplaceAllObjectsWithRequestOptions is the same as ReplaceAllObjects but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllObjectsWithRequestOptions(objects []Object, opts *RequestOptions) (err error) {
```

Comment:
Should we add the possibility of having a `safe` parameter?

Reply:
As safe is not handled at all so far and I'm planning on starting to integrate it in the coming next major, I would again delay this in the next major. Let me know if you think otherwise @Ant-hem @nunomaduro, there's no technical blocking point to implement it now.

Reply:
As discussed, because the change is breaking the compatibility (with regards to `RequestOptions`) and because we prefer to implement a new design for the `RequestOptions` in the next major release, we will not handle the optional `safe` parameter in this PR.